cloudinit/config/cc_users_groups.py

```diff
@@ -141,7 +141,7 @@
         ssh_import_id: [chad.smith]
         user:
           name: mynewdefault
-          sudo: false
+          sudo: null
```

Comment:
Wow this is a bug I introduced in a63f45f718. Cloud-init doesn't handle this deprecation or warn about it ... and it gets a traceback when launching with this suggested content.

```bash
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/cloudinit/config/modules.py", line 246, in _run_modules
    ran, _r = cc.run(
  File "/usr/lib/python3/dist-packages/cloudinit/cloud.py", line 67, in run
    return self._runners.run(name, functor, args, freq, clear_on_fail)
  File "/usr/lib/python3/dist-packages/cloudinit/helpers.py", line 185, in run
    results = functor(*args)
  File "/usr/lib/python3/dist-packages/cloudinit/config/cc_users_groups.py", line 206, in handle
    cloud.distro.create_user(user, **config)
  File "/usr/lib/python3/dist-packages/cloudinit/distros/__init__.py", line 655, in create_user
    self.write_sudo_rules(name, kwargs["sudo"])
  File "/usr/lib/python3/dist-packages/cloudinit/distros/__init__.py", line 814, in write_sudo_rules
    raise TypeError(msg % (type_utils.obj_name(rules)))
TypeError: Can not create sudoers rule addition with type 'NoneType'
```



Here's a logic fix that will also emit warnings on deprecated value:
```diff
diff --git a/cloudinit/distros/__init__.py b/cloudinit/distros/__init__.py
index ffa41093f..f2c744440 100644
--- a/cloudinit/distros/__init__.py
+++ b/cloudinit/distros/__init__.py
@@ -651,8 +651,15 @@ class Distro(persistence.CloudInitPickleMixin, metaclass=abc.ABCMeta):
             self.lock_passwd(name)
 
         # Configure sudo access
-        if "sudo" in kwargs and kwargs["sudo"] is not False:
-            self.write_sudo_rules(name, kwargs["sudo"])
+        if "sudo" in kwargs:
+            if kwargs["sudo"]:
+                self.write_sudo_rules(name, kwargs["sudo"])
+            elif kwargs["sudo"] is False:
+                LOG.warning(
+                    "DEPRECATED: The user %s has a 'sudo' config value of"
+                    " 'false' which is deprecated and will be removed in a"
+                    " future version. Use 'null' instead."
+                )
 
         # Import SSH keys
         if "ssh_authorized_keys" in kwargs:
```

Reply:
I pushed a commit c60e194b931ac88abd48c8887f6e77c0db185a64 on top of this PR for ease of your review. If undesireable, feel free to pop it off and refactor what you want.